event-management-backend/src/main/java/ro/msg/event_management/controller/EventController.java

```diff
@@ -311,5 +319,311 @@ public ResponseEntity options(HttpServletResponse response) {
+                continue;
+            }
+            event.setObservations(observations);
+
+            if(ticketsPerUser.equals("")) {
+                errorString += "Skipped invalid row\n";
+                continue;
+            }
+            event.setTicketsPerUser(Integer.parseInt(ticketsPerUser));
+
+            if(creator.equals("")) {
+                errorString += "Skipped invalid row\n";
+                continue;
+            }
+            event.setCreator(creator);
+
+            if(ticketInfo.equals("")) {
+                errorString += "Skipped invalid row\n";
+                continue;
+            }
+            event.setTicketInfo(ticketInfo);
+
+            if(locationId.equals("")){
+                errorString += "Skipped invalid row\n";
+                continue;
+            }
+
+            //add category
+            var categories = fields[16];
+            var ids4 = categories.split(";");
+            var list = new ArrayList<TicketCategory>();
+            for(String category: ids4){
+                var data = category.split("\\|");
+                var cat = new TicketCategory();
+                cat.setTitle(data[0]);
+                cat.setSubtitle(data[1]);
+                cat.setPrice(Float.parseFloat(data[2]));
+                cat.setDescription(data[3]);
+                cat.setTicketsPerCategory(Integer.parseInt(data[4]));
+                cat.setAvailable(Boolean.parseBoolean(data[5]));
+
+                list.add(cat);
+            }
+            event.setTicketCategories(list);
+
+            try {
+                Long.parseLong(locationId);
+            }catch (NumberFormatException e){
+                errorString += "Location id must be a number!\n";
+            }
+            event = eventService.saveEvent(event, Long.parseLong(locationId));
+
+            //add pictures to db
+            var urls = picturesUrls.split(";");
+            for(String url: urls){
+                var pic = new Picture();
+                pic.setEvent(event);
+                pic.setUrl(url);
+                pictureService.savePicture(pic);
+            }
+
+            //add eventsublocation to db
+            var sublocationIds = fields[15];
+            var ids2 = sublocationIds.split(";");
+            for(String id: ids2){
+                if(!id.equals("")) {
+                    try {
+                        var subLocation = sublocationService.findById(Long.parseLong(id));
+                        var eventSublocation = new EventSublocation();
+                        eventSublocation.setEvent(event);
+                        eventSublocation.setSublocation(subLocation);
+                        eventSublocationService.saveES(eventSublocation);
+                    }
+                    catch (Exception e){
+
+                    }
+                }
+            }
+        }
+
+        if(errorString.equals("")){
+            errorString = "Success";
+        }
```

Comment:
You should use StringBuffer for errorString instead of String.
Usually Controllers are cleaner from the logic perspective. I would like that most if not all of this logic is moved to EventService.